File: app/src/main/java/me/darshpratap/coaster/database/dao/ContentDao.java
package me.darshpratap.coaster.database.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Update;
import me.darshpratap.coaster.models.db.Content;

@Dao
public interface ContentDao {
    @Insert
    void insert(Content category);
    @Delete
    void delete(Content category);
    @Update
    void update(Content category);
}

File: app/src/main/java/me/darshpratap/coaster/database/dao/ContentDao.kt
package me.darshpratap.coaster.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import me.darshpratap.coaster.models.db.Content

@Dao
interface ContentDao {
    @Insert
    fun insert(content: Content)

    @Query("SELECT * FROM content WHERE cat_id=:id")
    fun getContentsForCategory(id: Int): List<Content>
}